test/jdk/com/sun/crypto/provider/KDF/HKDFKnownAnswerTests.java

```diff
@@ -0,0 +1,292 @@
+            }
+            expectedPRK = HexFormat.of().parseHex(Objects.requireNonNull(expPrkStr));
+            expectedOKM = HexFormat.of().parseHex(Objects.requireNonNull(expOkmStr));
+
+            // Non-mandatory fields - may be null
+            salt = (saltStr != null) ? HexFormat.of().parseHex(saltStr) : null;
+            info = (infoStr != null) ? HexFormat.of().parseHex(infoStr) : null;
+        }
+
+        public final String testName;
+        public final String algName;
+        public final byte[] ikm;
+        public final byte[] salt;
+        public final byte[] info;
+        public final int outLen;
+        public final byte[] expectedPRK;
+        public final byte[] expectedOKM;
+    }
+
+    public static final List<TestData> testList = new LinkedList<TestData>() {{
+        add(new TestData("RFC 5869 Test Case 1", "HKDF-SHA256",
+                         "0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b",
+                         "000102030405060708090a0b0c",
+                         "f0f1f2f3f4f5f6f7f8f9",
+                         42,
+                         "077709362c2e32df0ddc3f0dc47bba6390b6c73bb50f9c3122ec844ad7c2b3e5",
+                         "3cb25f25faacd57a90434f64d0362f2a2d2d0a90cf1a5a4c5db02d56ecc4c5bf"
+                         +
+                         "34007208d5b887185865"));
+        add(new TestData("RFC 5869 Test Case 2", "HKDF-SHA256",
+                         "000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"
+                         +
+                         "202122232425262728292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f"
+                         +
+                         "404142434445464748494a4b4c4d4e4f",
+                         "606162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f"
+                         +
+                         "808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9f"
+                         +
+                         "a0a1a2a3a4a5a6a7a8a9aaabacadaeaf",
+                         "b0b1b2b3b4b5b6b7b8b9babbbcbdbebfc0c1c2c3c4c5c6c7c8c9cacbcccdcecf"
+                         +
+                         "d0d1d2d3d4d5d6d7d8d9dadbdcdddedfe0e1e2e3e4e5e6e7e8e9eaebecedeeef"
+                         +
+                         "f0f1f2f3f4f5f6f7f8f9fafbfcfdfeff",
+                         82,
+                         "06a6b88c5853361a06104c9ceb35b45cef760014904671014a193f40c15fc244",
+                         "b11e398dc80327a1c8e7f78c596a49344f012eda2d4efad8a050cc4c19afa97c"
+                         +
+                         "59045a99cac7827271cb41c65e590e09da3275600c2f09b8367793a9aca3db71"
+                         +
+                         "cc30c58179ec3e87c14c01d5c1f3434f1d87"));
+        add(new TestData("RFC 5869 Test Case 3", "HKDF-SHA256",
+                         "0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b",
+                         new String(new byte[0]), new String(new byte[0]), 42,
```

Comment:
Instead of `new String(new byte[0])`, you should be able to just use `null` for optional arguments?